src/bin/sage-env

```diff
@@ -370,20 +370,31 @@ if [ -n "$PYTHONHOME" ]; then
 fi
 
 if [ -n "$SAGE_LOCAL" ]; then
-    # On OS X, test whether "ld-classic" is present in the installed
-    # version of the command-line tools. If so, we add "-ld_classic"
-    # to LD_FLAGS. See #36599.
-    # When the full XCode is installed and in use, for example after
-    # "sudo xcode-select -s /Applications/Xcode.app", then "xcode-select -p"
-    # gives "/Applications/Xcode.app/Contents/Developer", but "ld-classic"
-    # is not in the subdirectory "usr/bin/" but rather in the subdirectory
-    # "Toolchains/XcodeDefault.xctoolchain/usr/bin/". See #37237.
-    # However, if LD is set explicitly, as it is within conda on macOS,
-    # do not not do this.
-    if [ "$UNAME" = "Darwin" ] && [ -z "$LD" ] && [ -x "$(xcode-select -p)/usr/bin/ld-classic" -o -x "$(xcode-select -p)/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld-classic" ] ; then
-        LDFLAGS="-L$SAGE_LOCAL/lib -Wl,-ld_classic,-rpath,$SAGE_LOCAL/lib $LDFLAGS"
-    else
-        LDFLAGS="-L$SAGE_LOCAL/lib -Wl,-rpath,$SAGE_LOCAL/lib $LDFLAGS"
+    # On OS X, test whether "ld-classic" is present in the selected XCode
+    # toolchain. If so, add "-Wl,-ld_classic" to LDFLAGS (see #36599) unless
+    # LD is already set, as it will be with conda on macOS.  When the selected
+    # toolchain is in the Xcode app the output of "xcode-select -p" is
+    # "/Applications/Xcode.app/Contents/Developer", but "ld-classic" is not in
+    # the subdirectory "usr/bin/" but rather in the subdirectory
+    # "Toolchains/XcodeDefault.xctoolchain/usr/bin/".  (See #37237.)
+    LDFLAGS="-L$SAGE_LOCAL/lib"
```

Comment:
This change removes any LDFLAGS that may come from the user's environment. Is this intended?

Reply:
Good point!  No, that was not intended and seems like a bad idea.

Reply:
I pushed a fix.